### PR TITLE
BUG/REGR: Fix subset for DataFrameGroupBy.value_counts

### LINKED ISSUE #46383: BUG: subset parameter of DataFrameGroupBy.value_counts has no effect
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
>>> df = pd.DataFrame({"c1": ["a", "b", "c"], "c2": ["x", "y", "y"]}, index=[0, 1, 1])
>>> df
  c1 c2
0  a  x
1  b  y
1  c  y
>>> df.groupby(level=0).value_counts(subset=["c2"])
   c1  c2
0  a   x     1
1  b   y     1
   c   y     1
dtype: int64                                                                                                             dtype: int64
```

### Issue Description

`subset` has no effect. Within the code, there is only a check that subset and grouping keys do not intersect.

### Expected Behavior

Keeps only `subset` columns for counting unique combinations and drop the other columns.

```python
>>> df.groupby(level=0).value_counts(subset=["c2"])
   c2
0  x     1
1  y     2
dtype: int64
```

So it is consistent with `DataFrame.value_counts`:
```python
>>> df.value_counts(subset=["c2"])
c2
y     2
x     1
dtype: int64
```

In practice, this is equivalent of using `df.groupby(level=0)[["c2"]].value_counts()`.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 06d230151e6f18fdb8139d09abf539867a8cd481
python           : 3.9.1.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19041
machine          : AMD64
processor        : Intel64 Family 6 Model 142 Stepping 11, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : fr_FR.cp1252

pandas           : 1.4.1
numpy            : 1.20.1
pytz             : 2021.1
dateutil         : 2.8.1
pip              : 20.3.3
setuptools       : 52.0.0.post20210125
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.0.3
IPython          : 8.1.1
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : None
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None

</details>

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #46383 (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/v1.4.4.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -1805,21 +1805,24 @@ def value_counts(
                 name = self._selected_obj.name
                 keys = [] if name in in_axis_names else [self._selected_obj]
             else:
+                if subset is not None:
```
Comment (phofl) [inline pandas/core/groupby/generic.py:1808]: 2 things: What happens if a column from subset is not in the DataFrame at all?
Can we add a test with duplicated column names? Shouldn't be an issue with your implementation, but should help in avoiding potential regressions

Comment (mroeschke) [reply pandas/core/groupby/generic.py:1809]: > What happens if a column from subset is not in the DataFrame at all?

No exception would be raised & would act like there was no subset. Changed this to raising a `ValueError`.

> Can we add a test with duplicated column names? Shouldn't be an issue with your implementation, but should help in avoiding potential regressions

Added

**Review by phofl — APPROVED** [review]
Lgtm

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (phofl) [reply pandas/core/groupby/generic.py:1809]: Thx

File: doc/source/whatsnew/v1.4.4.rst [hunk 2]
```diff
@@ -37,6 +37,7 @@ Bug fixes
 ~~~~~~~~~
 - The :class:`errors.FutureWarning` raised when passing arguments (other than ``filepath_or_buffer``) as positional in :func:`read_csv` is now raised at the correct stacklevel (:issue:`47385`)
 - Bug in :meth:`DataFrame.to_sql` when ``method`` was a ``callable`` that did not return an ``int`` and would raise a ``TypeError`` (:issue:`46891`)
+- Bug in :meth:`DataFrameGroupBy.value_counts` where ``subset`` had no effect (:issue:`44267`)
```
Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.4.4.rst:40]: will be changing issue number to #46383 in #48314

Comment (mroeschke) [reply doc/source/whatsnew/v1.4.4.rst:40]: Thanks @simonjayhawkins! Sounds good

**Pull request merged**